### PR TITLE
[Token] fix issues

### PR DESCRIPTION
**Pull request by areshand**
### Description

1. Update one error code 
2. update the burn function to work with unlimited collection and token whose maximal is 0.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4316)

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
since this seems to be catching a bug, it would be great to add tests for these type of conditions

**Comment by areshand** [comment]
> since this seems to be catching a bug, it would be great to add tests for these type of conditions

added test and add test for some other public or entry function that wasn't covered sufficiently

File: aptos-move/framework/aptos-token/sources/token.move [hunk 1]
```diff
@@ -1483,4 +1488,79 @@ module aptos_token::token {
 
         withdraw_with_capability(cap)
     }
+
+    #[test(creator=@0xcafe, another_creator=@0xde)]
+    fun test_burn_token_from_both_limited_and_unlimited(
```
Comment (movekevin) [inline aptos-move/framework/aptos-token/sources/token.move:1493]: Can you separate this into 2 tests? I don't see why these 2 flows need to be in the same tests.

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-token/sources/token.move (hunk 1 above)
Comment (areshand) [reply aptos-move/framework/aptos-token/sources/token.move:1493]: This is to reduce the boilerplate code.  too many tests now.

**Review by chloeqjz — APPROVED** [review]

File: aptos-move/framework/aptos-token/sources/token.move [hunk 2]
```diff
@@ -1046,21 +1046,26 @@ module aptos_token::token {
...
-                royalty: _,
-                name: _,
-                description: _,
-                default_properties: _,
-                mutability_config: _,
-            } = table::remove(&mut collections.token_data, token_id.token_data_id);
+
+        // only update the supply if we tracking the supply and maximal
+        // maximal == 0 is reserved for unlimited token and collection with no tracking info.
+        if (token_data.maximum > 0) {
```
Comment (CapCap) [inline aptos-move/framework/aptos-token/sources/token.move:1052]: what if we want no maximum AND we want to track total supply?

**Pull request merged**